src/multiplex/client.rs

```diff
@@ -34,14 +34,80 @@ pub trait TagStore<Request, Response> {
     fn finish_tag(self: Pin<&mut Self>, r: &Response) -> Self::Tag;
 }
 
+pub trait PendingStore<T, Request>
+where
+    T: TryStream + Sink<Request> + TagStore<Request, T::Ok>,
```

Comment:
To return an `Error` it does since `Error<I>` has a `Sink<Request> + TryStream` bound unfortunately. Might be able to get away with some fancy `Into<BoxError>` bounds rather than explicit types.

Reply:
Or just have it be generic over `<Request, Response, E>` and do the association with `T` in the higher-level structures that contain `PendingStore`.

Reply:
To return an error we need the generic, no? https://github.com/tower-rs/tokio-tower/blob/master/src/error.rs#L8 Since `Error` expects some `T: Sink + Stream`. One option would be to maybe extract the generics but that seems like a massive refactor that would actually make some bounds longer (aka more generics where as we can combine them via this T)